core/src/main/java/org/elasticsearch/index/mapper/core/NumberFieldMapper.java

```diff
@@ -317,8 +318,14 @@ public void merge(Mapper mergeWith, MergeResult mergeResult) throws MergeMapping
         if (!this.getClass().equals(mergeWith.getClass())) {
             return;
         }
-        if (!mergeResult.simulate()) {
-            NumberFieldMapper nfmMergeWith = (NumberFieldMapper) mergeWith;
+        NumberFieldMapper nfmMergeWith = (NumberFieldMapper) mergeWith;
+        if (this.fieldTypeRef.getRefCount() > 1 && mergeResult.updateAllTypes() == false) {
```

Comment:
yeah if we name `getRefCount` `getNumAssociatedTypes` this line is way more intuitive to read?